### PR TITLE
Fix WrapperOpaque max encded len and type info

### PR DESCRIPTION
**Pull request by thiolliere**

### CODE REVIEW

**Comment by thiolliere** [comment]
@ascjones maybe you can review this one ? I forgot to declare the correct type info of the type

File: frame/support/src/traits/misc.rs [hunk 1]
```diff
@@ -425,6 +425,28 @@ impl<T> From<T> for WrapperOpaque<T> {
 	}
 }
 
+impl<T: MaxEncodedLen> MaxEncodedLen for WrapperOpaque<T> {
+	fn max_encoded_len() -> usize {
+		// Note: this can be improved if `T::max_encoded_len` is small.
+		// E.g. if T max encoded len is 4, then the compact encoding of its encoded length is 1.
```
Comment (bkchr) [inline frame/support/src/traits/misc.rs:431]: Why this comment here?

The point of this is to return the max encoded length?

**Review by bkchr — APPROVED** [review]

File: frame/support/src/traits/misc.rs (hunk 1 above)
Comment (thiolliere) [reply frame/support/src/traits/misc.rs]: yes but I think we can be more precise in certain situation, I finally did the more precise implementation https://github.com/paritytech/substrate/pull/9881/commits/5f0718626783285161a28a87189ff586b4fe3b36

**Review by bkchr — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Comment by thiolliere** [comment]
ah sorry I didn't see you asked review of ascjones, I'll ping him and ensure he get to see it

File: frame/support/src/traits/misc.rs [hunk 2]
```diff
@@ -425,6 +425,37 @@ impl<T> From<T> for WrapperOpaque<T> {
...
+
+impl<T: TypeInfo + 'static> TypeInfo for WrapperOpaque<T> {
+	type Identity = Self;
+	fn type_info() -> Type {
+		Type::builder()
+			.path(Path::new("WrapperOpaque", module_path!()))
+			.type_params(vec![TypeParameter::new("T", Some(meta_type::<T>()))])
+			.composite(
+				Fields::unnamed()
+					.field(|f| f.compact::<u32>().type_name("EncodedLengthOfT"))
```
Comment (ascjones) [inline frame/support/src/traits/misc.rs:453]: We don't need `type_name` here. It is optional and is used for the name of the type of the field as it appears in the code, in order to identify alias usage. E.g. where the concrete type is `u128` the `type_name` would be `Balance` for example.

**Review by ascjones — COMMENTED** [review]
LGTM. This type is potentially useful and universal enough to promote to `parity-scale-codec` and `scale-info` as a built-in.

File: frame/support/src/traits/misc.rs (hunk 2 above)
Comment (thiolliere) [reply frame/support/src/traits/misc.rs:453]: I see, I'll do another PR with this removed

**Pull request merged**